src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -1577,25 +1577,288 @@ class QuantizedOr(QuantizedOpUnivariateOfEncrypted, QuantizedOp):
+        new_scale = q_input_0.quantizer.scale * self.divider_quantizer.scale
+        new_zero_point = q_input_0.quantizer.zero_point * self.divider_quantizer.zero_point
+
+        if self.produces_graph_output:
+            return self.make_output_quant_parameters(product_q_values, new_scale, new_zero_point)
+
+        with tag(self.op_instance_name + ".rounding"):
+            # Apply Concrete rounding (if relevant)
+            product_q_values = self.cnp_round(product_q_values, calibrate_rounding)
+
+        # De-quantize the product
+        dequant_product = (product_q_values - new_zero_point) * new_scale
+
+        # Return the raw float values without re-quantizing them to the new scale, as any
+        # following Gemm/Add/Conv will quantize them with _prepare_inputs_with_constants(...)
+        return QuantizedArray(
+            self.n_bits,
+            dequant_product,
+            value_is_float=True,
+            options=self._get_output_quant_opts(),
+            stats=self.output_quant_stats,
+            params=self.output_quant_params,
+        )
+
+    def dump_dict(self) -> Dict:
+        metadata = super().dump_dict()
+        if hasattr(self, "min_non_zero_value"):
+            metadata["min_non_zero_value"] = self.min_non_zero_value
+        if hasattr(self, "divider_quantizer"):
+            metadata["divider_quantizer"] = self.divider_quantizer
+        return metadata
+
+    @staticmethod
+    def load_dict(metadata: Dict):
+        obj = super(QuantizedDiv, QuantizedDiv).load_dict(metadata)
+        if "min_non_zero_value" in metadata:
+            obj.min_non_zero_value = metadata.pop("min_non_zero_value")
+        if "divider_quantizer" in metadata:
+            obj.divider_quantizer = metadata.pop("divider_quantizer")
+        return obj
+
+    def can_fuse(self) -> bool:
+        """Determine if this op can be fused.
+
+        Add operation can be computed in float and fused if it operates over inputs produced
+        by a single integer tensor.
+
+        Returns:
+            bool: Whether the number of integer input tensors allows computing this op as a TLU
+        """
+
+        return len(self._int_input_names) == 1
+
 
-class QuantizedMul(QuantizedOpUnivariateOfEncrypted, QuantizedOp):
-    """Multiplication operator.
+class QuantizedMul(QuantizedMixingOp):
+    """Quantized Multiplication operator.
 
-    Only multiplies an encrypted tensor with a float constant for now. This operation will
-    be fused to a (potentially larger) TLU.
+    Can multiply either two variables (both encrypted) or a variable and a constant
     """
 
     _impl_for_op_named: str = "Mul"
 
+    def q_impl(
+        self,
+        *q_inputs: ONNXOpInputOutputType,
+        calibrate_rounding: bool = False,
+        **attrs,
+    ) -> ONNXOpInputOutputType:
+
+        # If either input is a RawOpOutput or if the op can be fused,
+        # perform the op in the TLU using FP32
```

Comment:
same as above

Reply:
same as above, why couldn't we do : 
```
if any(isinstance(input, RawOpOutput) for input in q_inputs) or self.can_fuse():
```

Reply:
I will clarify what's going on here. But I think it's due to the fact that we check QuantizedOps without calls to the `post_training.py` which is where we define the `can_fuse`. So I probably need to modify some test if I want to use a simple condition for all qops.

Reply:
not sure what you're saying, my observation was that here you basically do 
```
if len(q_inputs) == 1 or any(isinstance(input, RawOpOutput) for input in q_inputs) or self.can_fuse():
``` 
instead of 
```
if any(isinstance(input, RawOpOutput) for input in q_inputs) or self.can_fuse():
```
like above

so more or less : is this `len(q_inputs) == 1` necessary ?

Reply:
Let's see what the test say.

Reply:
I changed to only use can fuse. Seems to work fine.